core/templates/components/state-editor/state-hints-editor/state-hints-editor.component.spec.ts

```diff
@@ -242,58 +260,63 @@ describe('StateHintsEditorComponent', () => {
     expect($uibModal.open).not.toHaveBeenCalled();
   });
 
-  it('should open add hints modal when user clicks on add hint button', () => {
-    $scope.StateHintsService.displayed = [
-      {
-        hintContent: {
-          html: '<p> Hint </p>'
-        }
-      }
-    ];
-    spyOn($uibModal, 'open').and.returnValue({
-      result: $q.resolve({
-        hint: {
+  it('should open add hints modal when user clicks on add hint button',
+    fakeAsync(() => {
+      $scope.StateHintsService.displayed = [
+        {
           hintContent: {
-            html: '<p> New hint </p>'
+            html: '<p> Hint </p>'
           }
         }
-      })
-    });
+      ];
+      spyOn(ngbModal, 'open').and.returnValue({
+        componentInstance: new MockNgbModalRef(),
+        result: $q.resolve({
+          hint: {
+            hintContent: {
+              html: '<p> New hint </p>'
+            }
+          }
+        })
+      } as NgbModalRef);
 
-    $scope.openAddHintModal();
-    $scope.$apply();
+      $scope.openAddHintModal();
+      tick();
+      $scope.$apply();
```

Comment:
Is `$scope.$apply` necessary even after using a mock here and in the next it block?

Reply:
Removed.